sql/core/src/main/scala/org/apache/spark/sql/execution/dynamicpruning/PartitionPruning.scala

```diff
@@ -51,7 +56,7 @@ object PartitionPruning extends Rule[LogicalPlan] with PredicateHelper with Join
   /**
    * Search the partitioned table scan for a given partition column in a logical plan
    */
-  def getPartitionTableScan(a: Expression, plan: LogicalPlan): Option[LogicalRelation] = {
+  def getPartitionTableScan(a: Expression, plan: LogicalPlan): Option[LogicalPlan] = {
```

Comment:
nit: we might need to update the method name and doc because they are no longer accurate.